catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/ValidationFilterFactory.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.metacard.validation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.delegate.ValidationQueryDelegate;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.UnsupportedQueryException;
+
+public class ValidationFilterFactory {
+
+    private FilterAdapter adapter;
+
+    private FilterBuilder builder;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidationFilterFactory.class);
+
+    public ValidationFilterFactory(FilterAdapter filterAdapter, FilterBuilder filterBuilder) {
+        adapter = filterAdapter;
+        builder = filterBuilder;
+    }
+
+    public QueryRequest getFilter(QueryRequest input, boolean showInvalidMetacards) {
+        try {
+            if (showInvalidMetacards || adapter.adapt(input.getQuery(),
+                    new ValidationQueryDelegate())) {
+                return input;
+            }
+
+        } catch (UnsupportedQueryException e) {
+            LOGGER.error("This attribute filter is not supported by ValidationQueryDelegate.", e);
+        }
+        QueryImpl query = new QueryImpl(builder.allOf(input.getQuery(),
+                builder.attribute(BasicTypes.VALIDATION_ERRORS)
+                        .is()
+                        .empty(),
+                builder.attribute(BasicTypes.VALIDATION_WARNINGS)
+                        .is()
+                        .empty()));
+        query.setPageSize(input.getQuery()
+                .getPageSize());
+        query.setRequestsTotalResultsCount(input.getQuery()
+                .requestsTotalResultsCount());
+        input = new QueryRequestImpl(query,
```

Comment:
There's no value in setting the `input` handle to the object on this line and then returning it on the next. If a caller were to choose to look at the object they'd passed in instead of the object you're returning, they'd see the original, unreplaced object. Just change this to a `return new QueryRequestImpl...`